luigi/contrib/spark.py

```diff
@@ -378,6 +378,9 @@ def run(self):
         py_files = self.py_files()
         if py_files != []:
             options += ['--py-files', ','.join(py_files)]
+        dependency_jars = self.dependency_jars()
+        if dependency_jars != []:
```

Comment:
I agree @berkerpeksag.

There's also room for a bit more refactoring as ATM the run methods in PySpark1xJob and Spark1xJob are mostly duplicated.

I can have a go at it in separate PR?